docs/topics/db/aggregation.txt

```diff
@@ -643,6 +643,54 @@ aggregate that author count, referencing the annotation field:
     >>> Book.objects.annotate(num_authors=Count("authors")).aggregate(Avg("num_authors"))
     {'num_authors__avg': 1.66}
 
+When you are generating an aggregate by using ``.values()`` clause to create the
+grouping for the annotation on MySQL, and the ``ONLY_FULL_GROUP_BY`` SQL mode
+is enabled, it may be necessary to apply :class:`~django.db.models.AnyValue` if
+the annotation includes a mix of aggregated and non-aggregated functions.
+
+Take the following example:
+
+.. code-block:: pycon
+
+    >>> from django.db.models import F, Count, Greatest
+    >>> Book.objects.values(greatest_pages=Greatest("pages", 600)).annotate(
+    ...     num_authors=Count("authors"),
+    ...     pages_per_author=F("greatest_pages") / F("num_authors"),
+    ... ).aggregate(Avg("pages_per_author"))
+
+This creates groups of books based on the SQL column ``GREATEST(pages, 600)``.
+One unique group consists of books with 600 pages or less, and other unique
+groups will consist of books with the same pages. The ``pages_per_author``
+involves a mix of aggregated and non-aggregated functions, because
+``greatest_page`` is a non-aggregated function, whereas the ``num_authors`` is
+an aggregated function.
+
+The column in the group by clause is the expression for ``greatest_pages``.
+MySQL is not able to determine that the ``greatest_pages`` referenced in
+``pages_per_author`` is functionally dependent on the column in the group by
+clause, and it will raise the
+``OperationalError: incompatible with sql_mode=only_full_group_by`` error.
```

Comment:
I wonder if we should use _might_ here as MySQL might get smarter about that in the future

```suggestion
clause, and it might raise a
``OperationalError: incompatible with sql_mode=only_full_group_by`` error.
```